src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/service/NonAssociationsMergeService.kt

```diff
@@ -0,0 +1,79 @@
+package uk.gov.justice.digital.hmpps.hmppsnonassociationsapi.service
+
+import jakarta.transaction.Transactional
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.hmppsnonassociationsapi.jpa.NonAssociation
+import uk.gov.justice.digital.hmpps.hmppsnonassociationsapi.jpa.repository.NonAssociationsRepository
+import uk.gov.justice.digital.hmpps.hmppsnonassociationsapi.jpa.NonAssociation as NonAssociationJPA
+
+@Service
+@Transactional
+class NonAssociationsMergeService(
+  private val nonAssociationsRepository: NonAssociationsRepository,
+) {
+
+  companion object {
+    val log: Logger = LoggerFactory.getLogger(this::class.java)
+  }
+
+  fun mergePrisonerNumbers(oldPrisonerNumber: String, newPrisonerNumber: String): List<NonAssociationJPA> {
+    log.info("Replacing prisoner number $oldPrisonerNumber to $newPrisonerNumber")
+
+    val updatedRecords = mutableListOf<NonAssociation>()
+
+    nonAssociationsRepository.findAllByFirstPrisonerNumber(oldPrisonerNumber)
+      .plus(nonAssociationsRepository.findAllBySecondPrisonerNumber(oldPrisonerNumber)).forEach { nonAssociation ->
+
+        log.debug("Looking at record ${nonAssociation.id}")
+        if (nonAssociation.firstPrisonerNumber == oldPrisonerNumber) {
+          nonAssociationsRepository.findByFirstPrisonerNumberAndSecondPrisonerNumber(
+            firstPrisonerNumber = newPrisonerNumber,
+            secondPrisonerNumber = nonAssociation.secondPrisonerNumber,
+          ).let {
+            if (it != null) {
+              nonAssociationsRepository.delete(nonAssociation)
```

Comment:
Do we need to publish a domain event when a non-association is deleted?

Reply:
omg. i just realised that syncing this will be a nightmare. because a merge comes from nomis, it'd have it's own method of sorting out the change to non-associations. but if we change ours and they change theirs, how on earth can we align the sync?

Reply:
also, something to bear in mind is that we may potentially delete an open non-association but leave a closed one, which may be dangerous.

Reply:
The merge will already have happened in NOMIS so not we don't sync
Like in incentive we just need to align to what NOMIS now has.  

I will test out an set of NA in NOMIS and perform a merge to observe what happens.